### PR TITLE
Refactor Change HTTP Library

### PR DESCRIPTION
**Pull request by gjovs**
#### Description
This pull request to implement a transition from Express to Fastify, along with the addition of test suites and the reimplementation of adapters. This change aims to enhance performance, scalability, and overall stability of our platform.

#### What type of PR contains? 

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [x] 🔥 Performance Improvements
- [x] 🧑‍💻 Code Refactor
- [x] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

### CODE REVIEW

**Review by lucassm02 — CHANGES_REQUESTED** [review]
@gjovs congratulations on the implementation! I believe it must have been a huge challenge. Please check my comments so that we can proceed with the approval of this PR.

File: src/presentation/middlewares/token/validate-token-middleware.ts [hunk 1]
```diff
@@ -20,6 +20,9 @@ export class ValidateTokenMiddleware implements Middleware {
   ): Middleware.Result {
     try {
       const authorization = httpRequest.headers.authorization as string;
+
+      if (!authorization) return unauthorized('FICA FRIO AI MEU CHAPA');
```
[the remark this answers is not in the document]
Comment (gjovs) [reply src/presentation/middlewares/token/validate-token-middleware.ts:24]: For sure, fell sorry for that, I was really happy testing the fastify! 😁

File: src/infra/http/utils/http-server/http-server.ts [hunk 2]
```diff
@@ -164,44 +185,31 @@ export class HttpServer {
...
-    ...middlewares: RouteMiddleware[] | ExpressRoute[] | Function[]
+    middleware?: RouteMiddleware | Function,
+    ...middlewares: RouteMiddleware[] | Function[]
   ): Promise<void>;
   public async routesDirectory(
     path: string,
-    arg1?: string | RouteMiddleware | ExpressRoute | Function | Route,
-    ...args: RouteMiddleware[] | ExpressRoute[] | Function[]
+    arg1?: string | RouteMiddleware | Function | Route,
+    ...args: RouteMiddleware[] | Function[]
```
[the remark this answers is not in the document]
Comment (gjovs) [reply src/infra/http/utils/http-server/http-server.ts:203]: It's a mistake, In some scenarios I was trying to save the data of routes to cut some parts of the process when reload.

File: src/infra/http/utils/http-server/http-server.ts [hunk 3]
```diff
@@ -164,44 +185,31 @@ export class HttpServer {
...
     const extensionsToSearch = ['.TS', '.JS'];
     const ignoreIfIncludes = ['.MAP.', '.SPEC.', '.TEST.'];
 
-    const baseUrl = typeof arg1 === 'string' ? arg1 : this.baseUrl;
-
-    const middlewares =
-      typeof arg1 !== 'string' && arg1 !== undefined && !(arg1 instanceof Route)
-        ? [arg1, ...args]
-        : args;
+    this.paths.push(path);
```
[the remark this answers is not in the document]
Comment (gjovs) [reply src/infra/http/utils/http-server/http-server.ts:208]: It's a mistake, In some scenarios I was trying to save the data of routes to cut some parts of the process when reload.

File: src/infra/http/utils/http-server/http-server.ts [hunk 4]
```diff
@@ -164,44 +185,31 @@ export class HttpServer {
...
 
     const files = readdirSync(path);
 
-    const route =
-      arg1 instanceof Route
-        ? arg1
-        : this.createRoute({ hidden: true, baseUrl, path: '' });
-
-    if (middlewares.length) {
```
[the remark this answers is not in the document]
Comment (gjovs) [reply src/infra/http/utils/http-server/http-server.ts:201]: The middlewares wasn't being injected  right there, the implementation it's only looking to the default function imported, I'm fixing that

**Review by lucassm02 — APPROVED** [review]
@gjovs thank you for the implementation. This change will greatly improve the capacity and performance of our web server!😄

**Pull request merged**